app/controllers/admin/api/service_contracts_controller.rb

```diff
@@ -21,14 +34,52 @@ def destroy
     respond_with(service_subscription.unsubscribe(service_contract))
   end
 
+  # Service Subscription Update
+  # PUT /admin/api/accounts/{account_id}/service_contracts/{id}.xml
+  def update
+    service = service_contract.issuer
+    new_plan = service.service_plans.find(service_contract_plan_id)
+    service_contract.change_plan!(new_plan)
+    respond_with(service_contract)
+  end
+
+  # Service Subscription Show
+  # GET /admin/api/accounts/:account_id/service_contracts/:id.xml
+  def show
+    respond_with service_contract
+  end
+
   protected
 
   def account
     @account ||= current_account.buyers.find params.require(:account_id)
   end
 
   def service_contract
-    @service_contract ||= account.bought_service_contracts.find params.require(:id)
+    @service_contract ||= account.bought_service_contracts.find_by(id: params.require(:id))
+  end
+
+
+  def service_contract_params
+    params.permit(service_contract: [:plan_id])
+          .fetch(:service_contract).merge(plan: service_plan)
   end
 
+  def service_plan
+    ServicePlan.find_by(id: service_contract_plan_id)
+  end
+
+  def create_service_contract(plan)
+    @service_contract = account.bought_service_contracts.create(service_contract_params.merge(plan: plan))
+    respond_with @service_contract
```

Comment:
Here we don't need to use this as an instance variable `@service_contract`, just a local one, e.g. `new_contract`
In fact:
1. We don't need an extra variable at all, and can just do 
```
respond_with account.bought_service_contracts.create(service_contract_params.merge(plan: plan))
```

2. We don't need this extra method, and can move this line inside the `#create`.

Reply:
And also, `service_contract_params` already merges in the `plan` instance, don't need to merge it in again.